### PR TITLE
fix(datepicker): keyboard fixes (enter matches click behavior & corre…

### LINKED ISSUE #7142: Datepicker - close calendar after selecting date using keyboard
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Datepicker calendar should close when current day or another day is selected via keyboard i.e. arrow keys + Enter.

#### What is the current behavior?
Datepicker calendar only closes when a date is selected using mouse.

#### What are the steps to reproduce?
Open the datepicker, use the arrow keys to select a date, press Enter. Datepicker calendar remains open and no selection is made.

The example datepicker exhibits the behaviour: https://material.angular.io/components/datepicker/examples

#### What is the use-case or motivation for changing an existing behavior?
Consistent behaviour with mouse vs keyboard. Better usability for users who traverse forms using a keyboard.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular v4, material 2.0.0-beta.10

### PR DESCRIPTION
**Pull request by mmalerba**
…ct up and down arrow behavior in year mode).

fixes #7142

### CODE REVIEW

**Comment by willshowell** [comment]
Does this fix https://github.com/angular/material2/issues/7142?

File: src/lib/datepicker/calendar.ts [hunk 1]
```diff
@@ -287,6 +287,7 @@ export class MdCalendar<D> implements AfterContentInit, OnDestroy {
       case ENTER:
         if (this._dateFilterForViews(this._activeDate)) {
           this._dateSelected(this._activeDate);
+          this._userSelected();
```
Comment (jelbourn) [inline src/lib/datepicker/calendar.ts]: Add test?

**Comment by mmalerba** [comment]
fixed existing tests and added test for new behavior

**Comment by tinayuangao** [comment]
@mmalerba please rebase. Thanks!

**Pull request merged**